packages/app-spec-squad-chess/src/index.js

```diff
@@ -1,174 +1,46 @@
-      ]
-    },
-    graphics: {
-      local: {
-        white: 'chesspieces/wikipedia/wR.png',
-        black: 'chesspieces/wikipedia/bR.png'
-      }
-    }
-  },
-  king: {
-    king: true,
-    mechanics: {
-      move: [
-        { offset: [0,1], steps: 1 },
-        { offset: [0,-1], steps: 1 },
-        { offset: [1,0], steps: 1 },
-        { offset: [-1,0], steps: 1 },
-        { offset: [1,1], steps: 1 },
-        { offset: [-1,-1], steps: 1 },
-        { offset: [1,-1], steps: 1 },
-        { offset: [-1,1], steps: 1 }
-      ],
-      capture: [
-        { offset: [0,1], steps: 1 },
-        { offset: [0,-1], steps: 1 },
-        { offset: [1,0], steps: 1 },
-        { offset: [-1,0], steps: 1 },
-        { offset: [1,1], steps: 1 },
-        { offset: [-1,-1], steps: 1 },
-        { offset: [1,-1], steps: 1 },
-        { offset: [-1,1], steps: 1 }
-      ]
-    },
-    graphics: {
-      local: {
-        white: 'chesspieces/wikipedia/wK.png',
-        black: 'chesspieces/wikipedia/bK.png'
-      }
-    }
+const App = {
+  view: () => {
+    return m(Board)
   }
 }
 
-let mockStartingPosition = {
-  '0,0': {
-    pieceId: 'pawn',
-    player: 0
-  },
-  '0,1': null,
-  '0,2': null,
-  '0,3': {
-    pieceId: 'knight',
-    player: 0
-  },
-  '1,0': {
-    pieceId: 'rook',
-    player: 0
-  },
-  '1,1': {
-    pieceId: 'king',
-    player: 0
-  },
-  '1,2': null,
-  '1,3': null,
-  '2,0': null,
-  '2,1': {
-    pieceId: 'pawn',
-    player: 1
-  },
-  '2,2': {
-    pieceId: 'king',
-    player: 1
-  },
-  '2,3': null,
-  '3,0': null,
-  '3,1': {
-    pieceId: 'knight',
-    player: 1
-  },
-  '3,2': {
-    pieceId: 'rook',
-    player: 1
-  },
-  '3,3': null
-}
-
-chess.registerPieces(mockPieceList)
-let turns = chess.generateTurns(mockStartingPosition, 0)
-
-state['game'] = {
-  position: mockStartingPosition,
-  turnNumber: 0,
-  legalTurns: turns
-}
-
-state['pieces'] = mockPieceList
+async function init() {
+  metastore.webSocketConnection(settings.metastoreWs)
+  const formatDefs = await metastore.getGameFormats(settings.gameAddress)
+  state.formats = formatDefs.map(def => def.Format)
+  const format = state.formats[0] // TODO build in a format selection interface
+  console.log(format)
+  const components = await Promise.all(
+    format.components.map(metastore.getDefinition)
+  )
+  const pieces = components.map(
+    c => JSON.parse(c.Component.data)
+  ).reduce((ps, p) => {
+    return Object.assign(ps, p)
+  })
+  chess.registerPieces(pieces)
+  const startingPosition = JSON.parse(format.data).startingPosition
+  let turns = chess.generateTurns(startingPosition, 0)
+
+  state['game'] = {
+    position: startingPosition,
+    turnNumber: 0,
+    legalTurns: turns
+  }
 
-// end initial state
+  state['pieces'] = pieces
 
-const App = {
-  view: () => {
-    return m(Board)
-  }
+  return "Squad Chess Initialized"
 }
 
-m.mount(document.body, App)
\ No newline at end of file
+init().then((message) => {
```

Comment:
I think this might be possible using the `init` mithril lifecycle function (making in `init` that goes above `view` in `App`, but let's just merge this for now, and I can mess with that in another PR.